### ISSUE TITLE
notification improvements to "ingest failure" message: provide better notification to users to avoid support tickets in RT regarding this feature

### ISSUE THREAD

**Issue by sbarbosadataverse** [IQSS/dataverse.harvard.edu#122]
Since this isn't a bug, provide better notification messages in the UI to users so they understand they don't need to contact us about failed ingests and they understand the failed to ingest message is not shown to users of their data.

**Comment by TaniaSchlatter** [comment]
It sounds like users need feedback about the status of their file @sbarbosadataverse, which is different from a warning or error.

**Comment by sbarbosadataverse** [comment]
@TaniaSchlatter thanks for chiming in. the message as it exists now, does not include reference as to why the failure is or is not important. users don't understand what it means and I think many assume their files are not "uploaded" when they see it

**Comment by TaniaSchlatter** [comment]
Based on the above @sbarbosadataverse, in the case of Stata ingest failure (the problem that prompted this issue), line breaks in Excel (https://github.com/IQSS/dataverse/issues/6874), Excel string values (https://github.com/IQSS/dataverse/issues/3383), or other (https://github.com/IQSS/dataverse/issues/7854), if the file is added (without features enabled by completed ingest), we should consider:

- not showing an error, warning, or other message in the file table on the dataset page
- not showing an error, warning, or other message on the file page
- using the in-app notification and email to the depositor to inform the user of the status of the dataset/file (that the file is available but that ingest was not completed), what ingest means, and requirements for ingest.

**Images for context**
Excel file with a string break, and other irregularities as far as ingest is concerned. 
_Propose removing this notification._
<img width="1174" alt="Screen Shot 2021-09-14 at 2 59 14 PM" src="https://user-images.githubusercontent.com/15386424/133319695-1aead36c-f219-462e-9602-b9001c7cb5f3.png">
<img width="1183" alt="Screen Shot 2021-09-14 at 2 59 27 PM" src="https://user-images.githubusercontent.com/15386424/133319699-df05461a-bfc0-49f8-944b-e9f4f0988b69.png">

**In-app notifications.**
_Propose keeping, but changing._
<img width="1220" alt="Screen Shot 2021-09-14 at 3 00 08 PM" src="https://user-images.githubusercontent.com/15386424/133319700-8a9a8c5a-16b1-4944-ae2c-aa707ccbd030.png">

**File page notification**
_Propose removing._
<img width="1227" alt="Screen Shot 2021-09-14 at 2 59 45 PM" src="https://user-images.githubusercontent.com/15386424/133319702-a336361e-07c9-4144-adad-453d137dc228.png">


Looking forward to hearing your thoughts @sbarbosadataverse as well as @djbrooke @scolapasta.

**Comment by djbrooke** [comment]
It's important IMHO that we have on page messaging to show why a file failed, on both the dataset and file page, as you could have multiple people working with a dataset and I'd like to surface it for all parties. The goal is to get people to at least see the error and determine if it's something they can/should fix. The ingest tool will never work for all files and there will always be failures. I do think the messages are overly persistent and not clear, but I'm not in favor of removing them. 

We should revisit notifications and emails, definitely! The failed ingest emails were added in the last few years by an external contributor if I recall correctly.

**Comment by sbarbosadataverse** [comment]
@djbrooke  I think @Leonid would be great at capturing the useful message for users if we keep the messaging. That's all we want is for users to understand what they are seeing. Leonid has helped me in the past with figuring out how a depositor should correct their file for successful ingest, but we have cases where file formats and versions shouldn't be changed so we don't want to encourage that. So the message has to be very clear. And in the case where ingest won't work for every file, I don't think the message should encourage changes to the files. But it should tell the user in so many words that "ingest failed, and your file won't have this "Additional" support of working with the data explorer tool. files will still be downloadable in original format by users. This message is only available to you. For additional information on tabular file ingests, follow this link."

**Comment by qqmyers** [comment]
In some sense, this isn't a failure per se. There are perfectly valid Excel /other files that don't conform to the expectation that they represent one table. Could this be an informative msg versus a warning/error (in addition to text changes)?

**Comment by sbarbosadataverse** [comment]
@qqmyers agree. it's not a failure. that's part 1 of cleaning up the message. part 2, is cleaning up the message well enough that users aren't sending us support tickets and repeatedly deleting their file and retrying the upload, with the same results, which is what is happening. 

@djbrooke  how about sending it in the notification under "my data" and leaving it there may help. we can certainly add more text to notifications that go under my data than on the user interface? what do you think? and I agree, to removing "failed" and using some other term to capturing what's happening.

**Comment by djbrooke** [comment]
@sbarbosadataverse I only have strong feelings regarding the removal of the on page messages, which I feel we should not do.  I'm supportive of toning the messages down and making it less persistent and/or dismissable. I'm supportive of providing information to users that will allow them to decide whether or not to take the time to get the file in a state that will allow it to be ingested or to get them information that will allow them to make the decision to be OK with the failed ingest because it's a table of contents or has unsupported characters or some other case. I'm very, very supportive of making any changes to the notifications/emails.

**Comment by sbarbosadataverse** [comment]
@djbrooke Ok. Then sounds like we can get Leonid on board to help us with that messaging? In case there is technical information we have to include? and where the messaging will be visible? Should we include some community feedback on this? I am sure Sebastian, Philipp may want to add some input...

**Comment by TaniaSchlatter** [comment]
**Proposed**
Revised, based on discussions in the issue:

**1. Show a message to depositor and others with access on the dataset page**
The [info style message](https://guides.dataverse.org/en/latest/style/patterns.html#alerts) is the UI affordance that is used for this type of situation:

<img width="867" alt="Screen Shot 2021-09-14 at 4 24 43 PM" src="https://user-images.githubusercontent.com/15386424/133331399-97cb5d7e-ea6e-492d-8490-20fe136bcdcb.png"> 

**2. Show the same info style message on the file page?** Is this useful?

**3. Use the current in-app notification and email** to the depositor to inform the user of the status of the dataset/file (that the file is available but that ingest was not completed), what ingest means, and requirements for ingest.

**4. We need to work out wording** that appears in these places. @sbarbosadataverse @djbrooke @scolapasta

**Comment by djbrooke** [comment]
Thx @TaniaSchlatter -

1. Yes, with "at least one file has failed ingest" - this introduces a potential issue where the person would have to sift through a large number of files to find the one that failed, but with the median # of files per dataset ~10 in Harvard Dataverse I think it'd be OK as it would generally appear on the first page.
2. Yes
3. Yes, we should update. We have more real estate here.
4. Yes
5. I think we should also beef up the part of the user guide that covers ingest, and link to it from these messages.
